core/src/main/java/io/questdb/griffin/engine/table/ShowCreateTableRecordCursorFactory.java

```diff
@@ -0,0 +1,273 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+package io.questdb.griffin.engine.table;
+
+import io.questdb.cairo.AbstractRecordCursorFactory;
+import io.questdb.cairo.CairoColumn;
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.CairoTable;
+import io.questdb.cairo.ColumnType;
+import io.questdb.cairo.GenericRecordMetadata;
+import io.questdb.cairo.MetadataCacheReader;
+import io.questdb.cairo.PartitionBy;
+import io.questdb.cairo.TableColumnMetadata;
+import io.questdb.cairo.TableToken;
+import io.questdb.cairo.sql.NoRandomAccessRecordCursor;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.RecordCursor;
+import io.questdb.cairo.sql.RecordMetadata;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.str.Path;
+import io.questdb.std.str.Utf8Sequence;
+import io.questdb.std.str.Utf8StringSink;
+import org.jetbrains.annotations.NotNull;
+
+public class ShowCreateTableRecordCursorFactory extends AbstractRecordCursorFactory {
+    public static final int N_DDL_COL = 0;
+    private static final RecordMetadata METADATA;
+    private final ShowCreateTableCursor cursor = new ShowCreateTableCursor();
+    private final TableToken tableToken;
+    private final int tokenPosition;
+
+    public ShowCreateTableRecordCursorFactory(TableToken tableToken, int tokenPosition) {
+        super(METADATA);
+        this.tableToken = tableToken;
+        this.tokenPosition = tokenPosition;
+    }
+
+    @Override
+    public RecordCursor getCursor(SqlExecutionContext executionContext) throws SqlException {
+        return cursor.of(executionContext, tableToken, tokenPosition);
+    }
+
+    @Override
+    public boolean recordCursorSupportsRandomAccess() {
+        return false;
+    }
+
+    @Override
+    public void toPlan(PlanSink sink) {
+        sink.type("show_create_table");
+        sink.meta("of").val(tableToken.getTableName());
+    }
+
+    public static class ShowCreateTableCursor implements NoRandomAccessRecordCursor {
+        private final ShowCreateTableRecord record = new ShowCreateTableRecord();
+        private final Utf8StringSink sink = new Utf8StringSink();
+
+        private SqlExecutionContext executionContext;
+        private boolean hasRun;
+        private CairoTable table;
+        private TableToken tableToken;
+
+        @Override
+        public void close() {
+            sink.clear();
+        }
+
+        @Override
+        public Record getRecord() {
+            return record;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (!hasRun) {
```

Comment:
this variable seems not used as intended - it is never set to `false`. This is also not tested (i am unsure we need testing like this) if `hasNext()` is called again after it returned `false` the first time (because it will do the same work)

Reply:
sorry, what i said is not true, variable is set to `true`, ignore this comment please